srcs/execute/env_utils.c

```diff
@@ -0,0 +1,102 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        ::::::::            */
+/*   env_utils.c                                        :+:    :+:            */
+/*                                                     +:+                    */
+/*   By: jsaariko <jsaariko@student.codam.nl>         +#+                     */
+/*                                                   +#+                      */
+/*   Created: 2020/10/28 11:19:34 by jsaariko      #+#    #+#                 */
+/*   Updated: 2020/10/29 13:51:00 by jsaariko      ########   odam.nl         */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "execute.h"
+#include "error.h"
+
+t_env		*get_env_item(char *env_str)
+{
+	t_env	*item;
+	int		i;
+
+	item = (t_env *)e_malloc(sizeof(t_env));
+	i = 0;
+	while (env_str[i] != '\0' && env_str[i] != '=')
+		i++;
+	item->key = (char *)e_malloc(sizeof(char) * (i + 1));
+	ft_strlcpy(item->key, env_str, i + 1);
+	if (env_str[i] != '=')
+		item->value = NULL;
+	else
+	{
+		item->value = ft_strdup(env_str + i + 1);
+		if (item->value == NULL)
+			error_exit_errno();
+	}
+	return (item);
+}
+
+t_vector	*envp_to_env(char **envp)
+{
+	size_t		i;
+	size_t		count;
+	t_env		*cur;
+	t_vector	*env;
+	int			ret;
```

Comment:
the naming convention that `cur` is a `t_env` but `env` is a `t_vector` could be slightly clearer haha but its fine xD

Reply:
given that t_env is a fairly generic struct (two strings of key/value pair), we could even call it `t_kvp` or something and store it somewhere more general so other things can use it too? Only if we do change the above though, but like I said it's fine as it is.